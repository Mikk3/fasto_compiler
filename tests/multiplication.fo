fun int main() =
  let x = read(int) in
  let y = read(int) in
  let out = write(x * y) in // 5 * 4 = 20
  let out = write(x * (0 - y)) in // 5 * -4 = -20
  let out = write((0 - x) * y) in // -5 * 4 = -20
  let out = write((0 - x) * (0 - y)) in // -5 * -4 = 20 
  let out = write(1 + 3 * 4) in // 1 + 3 * 4 = 13
  let out = write( 5 * 2 == 10 * 2) in // false (precedence tests)
  let out = write( 10 * 2 == 10 * 2) in // true
  0