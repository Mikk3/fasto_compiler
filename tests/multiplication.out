20 -20 -20 20 13 false true